crates/astria-sequencer-validation/src/proof.rs

```diff
@@ -0,0 +1,214 @@
+use ct_merkle::{
+    inclusion::InclusionProof as CtInclusionProof,
+    CtMerkleTree,
+    RootHash,
+};
+use serde::{
+    ser::SerializeStruct,
+    Deserialize,
+    Serialize,
+};
+use sha2::Sha256;
+use tendermint::Hash;
+
+/// A wrapper around [`ct_merkle::CtMerkleTree`], which uses sha256 as the hashing algorithm
+/// and Vec<u8> as the leaf type.
+#[derive(Debug, Default)]
+pub struct MerkleTree(CtMerkleTree<Sha256, Vec<u8>>);
+
+impl MerkleTree {
+    #[must_use]
+    pub fn from_leaves(leaves: Vec<Vec<u8>>) -> Self {
+        let tree = leaves
+            .into_iter()
+            .fold(CtMerkleTree::new(), |mut tree, leaf| {
+                tree.push(leaf);
+                tree
+            });
+        MerkleTree(tree)
+    }
+
+    #[must_use]
+    pub fn root(&self) -> Hash {
```

Comment:
It's very surprising that this returns tendermint types because the crate description does not mention it.

Can this return `[u8; 32]` instead?

Reply:
yes, updated to no longer depend on tendermint. it's now just a dev dependency since I have the compatibility unit tests